src/base/cub_ddl_log.c

```diff
@@ -0,0 +1,1047 @@
+static FILE *cub_fopen_and_lock (const char *path, const char *mode);
+
+void
+cub_ddl_log_init ()
+{
+  if (ddl_audit_handle != NULL)
+    {
+      FREE_MEM (ddl_audit_handle->sql_text);
+      FREE_MEM (ddl_audit_handle->err_msg);
+      FREE_MEM (ddl_audit_handle);
+    }
+
+  ddl_audit_handle = (T_DDL_AUDIT_HANDLE *) MALLOC (sizeof (T_DDL_AUDIT_HANDLE));
+  if (ddl_audit_handle == NULL)
+    {
+      return;
+    }
+
+  memset (ddl_audit_handle, 0x00, sizeof (T_DDL_AUDIT_HANDLE));
+
+  ddl_audit_handle->pid = 0;
+  ddl_audit_handle->stmt_type = -1;
+  ddl_audit_handle->br_index = 0;
+  ddl_audit_handle->file_line_number = 0;
+  ddl_audit_handle->err_code = 0;
+  ddl_audit_handle->log_type = DDL_LOG_ELAPSED_TIME;
+  ddl_audit_handle->loaddb_file_type = LOADDB_FILE_TYPE_NONE;
+}
+
+void
+cub_ddl_log_free ()
+{
+  if (ddl_audit_handle == NULL)
+    {
+      cub_ddl_log_init ();
+      return;
+    }
+
+  FREE_MEM (ddl_audit_handle->sql_text);
+  FREE_MEM (ddl_audit_handle->err_msg);
+
+  memset (ddl_audit_handle, 0x00, sizeof (T_DDL_AUDIT_HANDLE));
+
+  ddl_audit_handle->pid = 0;
+  ddl_audit_handle->stmt_type = -1;
+  ddl_audit_handle->file_line_number = 0;
+  ddl_audit_handle->err_code = 0;
+  ddl_audit_handle->log_type = DDL_LOG_ELAPSED_TIME;
+  ddl_audit_handle->loaddb_file_type = LOADDB_FILE_TYPE_NONE;
+}
+
+void
+cub_ddl_log_destroy ()
+{
+  if (ddl_audit_handle != NULL)
+    {
+      FREE_MEM (ddl_audit_handle->sql_text);
+      FREE_MEM (ddl_audit_handle->err_msg);
+      FREE_MEM (ddl_audit_handle);
+    }
+}
+
+void
+cub_ddl_log_app_name (const char *app_name)
+{
+  if (ddl_audit_handle == NULL || app_name == NULL)
+    {
+      return;
+    }
+
+  snprintf (ddl_audit_handle->app_name, sizeof (ddl_audit_handle->app_name), app_name);
+}
+
+void
+cub_ddl_log_db_name (const char *db_name)
+{
+  char *pstr = NULL;
+  if (ddl_audit_handle == NULL || db_name == NULL)
+    {
+      return;
+    }
+
+  snprintf (ddl_audit_handle->db_name, sizeof (ddl_audit_handle->db_name), db_name);
+
+  pstr = (char *) strchr (ddl_audit_handle->db_name, '@');
+  if (pstr != NULL)
+    {
+      *pstr = '\0';
+    }
+}
+
+void
+cub_ddl_log_user_name (const char *user_name)
+{
+  if (ddl_audit_handle == NULL || user_name == NULL)
+    {
+      return;
+    }
+
+  snprintf (ddl_audit_handle->user_name, sizeof (ddl_audit_handle->user_name), user_name);
+}
+
+void
+cub_ddl_log_ip (const char *ip_addr)
+{
+  if (ddl_audit_handle == NULL || ip_addr == NULL)
+    {
+      return;
+    }
+
+  snprintf (ddl_audit_handle->ip_addr, sizeof (ddl_audit_handle->ip_addr), ip_addr);
+}
+
+void
+cub_ddl_log_pid (const int pid)
+{
+  if (ddl_audit_handle == NULL)
+    {
+      return;
+    }
+
+  ddl_audit_handle->pid = pid;
+}
+
+void
+cub_ddl_log_br_name (const char *br_name)
+{
+  if (ddl_audit_handle == NULL || br_name == NULL)
+    {
+      return;
+    }
+  snprintf (ddl_audit_handle->br_name, 64, br_name);
```

Comment:
What about change it to:
`snprintf (ddl_audit_handle->br_name, BROKER_NAME_LEN, br_name);`